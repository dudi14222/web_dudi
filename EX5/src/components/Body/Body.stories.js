import React from 'react';
import { storiesOf } from '@kadira/storybook';
import {
  withKnobs
} from '@kadira/storybook-addon-knobs';

import Body from './';

const storiesBody = storiesOf('Body', module);
storiesBody.addDecorator(withKnobs);
storiesBody.addWithInfo('with children', '', () => {
  const story = <Body >
    <label>label</label>
  </Body>;
  return story;
});

