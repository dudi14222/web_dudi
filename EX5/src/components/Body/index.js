import React from 'react';
import './body.css';

const Body = ({ children }) => (
    <div className="col-md-12 app-body">
        { children }
    </div>
)
export default Body;