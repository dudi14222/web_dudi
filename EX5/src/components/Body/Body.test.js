import Body from '../Body';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";


describe('Body snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Body />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('Body shallow tests', () => {
    it('should render children when passed in', () => {
        const wrapper = shallow((
            <Body>
                <div className="unique" />
            </Body>
        ));    
        expect(wrapper.contains(<div className="unique" />)).toBe(true);
    });
})



