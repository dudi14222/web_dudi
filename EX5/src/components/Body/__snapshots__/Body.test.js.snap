// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body snapshot tests test snapshot 1`] = `
<div
  className="col-md-12 app-body"
/>
`;
